// @copilotkit/runtime/dist/index.mjs
import {
  CopilotRuntime,
  flattenToolCallsNoDuplicates
} from "./chunk-B6PJXTDZ.mjs";
import {
  AnthropicAdapter,
  GoogleGenerativeAIAdapter,
  GroqAdapter,
  LangChainAdapter,
  OpenAIAdapter,
  OpenAIAssistantAdapter,
  RemoteChain,
  UnifyAdapter
} from "./chunk-OF6AN6HF.mjs";
import {
  config,
  copilotRuntimeNextJSAppRouterEndpoint,
  copilotRuntimeNextJSPagesRouterEndpoint
} from "./chunk-KKBYD646.mjs";
import {
  copilotRuntimeNestEndpoint
} from "./chunk-3YMEAKGB.mjs";
import {
  copilotRuntimeNodeExpressEndpoint
} from "./chunk-XWY3EISO.mjs";
import {
  buildSchema,
  copilotRuntimeNodeHttpEndpoint,
  createContext,
  getCommonConfig
} from "./chunk-SWIWO5AZ.mjs";
import "./chunk-73NMP3DI.mjs";
import {
  GuardrailsValidationFailureResponse,
  MessageStreamInterruptedResponse,
  UnknownErrorResponse
} from "./chunk-U3V2BCGI.mjs";
import "./chunk-TBZGOJJX.mjs";
import "./chunk-RMZWGQ46.mjs";
import "./chunk-44O2JGUY.mjs";

// src/index.ts
import "reflect-metadata";
export {
  AnthropicAdapter,
  CopilotRuntime,
  GoogleGenerativeAIAdapter,
  GroqAdapter,
  GuardrailsValidationFailureResponse,
  LangChainAdapter,
  MessageStreamInterruptedResponse,
  OpenAIAdapter,
  OpenAIAssistantAdapter,
  RemoteChain,
  UnifyAdapter,
  UnknownErrorResponse,
  buildSchema,
  config,
  copilotRuntimeNestEndpoint,
  copilotRuntimeNextJSAppRouterEndpoint,
  copilotRuntimeNextJSPagesRouterEndpoint,
  copilotRuntimeNodeExpressEndpoint,
  copilotRuntimeNodeHttpEndpoint,
  createContext,
  flattenToolCallsNoDuplicates,
  getCommonConfig
};
