// @copilotkit/react-ui/dist/chunk-FOZVHL73.mjs
import {
  Markdown
} from "./chunk-YQ3D5IQV.mjs";
import {
  useChatContext
} from "./chunk-BJPGMY3I.mjs";

// src/components/chat/Messages.tsx
import React, { useEffect, useMemo } from "react";
import { useCopilotContext } from "@copilotkit/react-core";
import {
  MessageStatusCode,
  ActionExecutionMessage,
  ResultMessage,
  TextMessage,
  Role,
  AgentStateMessage
} from "@copilotkit/runtime-client-gql";
import { jsx, jsxs } from "react/jsx-runtime";
var Messages = ({ messages, inProgress, children }) => {
  const { chatComponentsCache } = useCopilotContext();
  const context = useChatContext();
  const initialMessages = useMemo(
    () => makeInitialMessages(context.labels.initial),
    [context.labels.initial]
  );
  messages = [...initialMessages, ...messages];
  const functionResults = {};
  for (let i = 0; i < messages.length; i++) {
    if (messages[i] instanceof ActionExecutionMessage) {
      const id = messages[i].id;
      const resultMessage = messages.find(
        (message) => message instanceof ResultMessage && message.actionExecutionId === id
      );
      if (resultMessage) {
        functionResults[id] = ResultMessage.decodeResult(resultMessage.result || "");
      }
    }
  }
  const messagesEndRef = React.useRef(null);
  const scrollToBottom = () => {
    if (messagesEndRef.current) {
      messagesEndRef.current.scrollIntoView({
        behavior: "auto"
      });
    }
  };
  useEffect(() => {
    scrollToBottom();
  }, [messages]);
  return /* @__PURE__ */ jsxs("div", { className: "copilotKitMessages", children: [
    messages.map((message, index) => {
      const isCurrentMessage = index === messages.length - 1;
      if (message instanceof TextMessage && message.role === "user") {
        return /* @__PURE__ */ jsx("div", { className: "copilotKitMessage copilotKitUserMessage", children: message.content }, index);
      } else if (message instanceof TextMessage && message.role == "assistant") {
        return /* @__PURE__ */ jsx("div", { className: `copilotKitMessage copilotKitAssistantMessage`, children: isCurrentMessage && inProgress && !message.content ? context.icons.spinnerIcon : /* @__PURE__ */ jsx(Markdown, { content: message.content }) }, index);
      } else if (message instanceof ActionExecutionMessage) {
        if (chatComponentsCache.current !== null && chatComponentsCache.current.actions[message.name]) {
          const render = chatComponentsCache.current.actions[message.name];
          if (typeof render === "string") {
            if (isCurrentMessage && inProgress) {
              return /* @__PURE__ */ jsxs("div", { className: `copilotKitMessage copilotKitAssistantMessage`, children: [
                context.icons.spinnerIcon,
                " ",
                /* @__PURE__ */ jsx("span", { className: "inProgressLabel", children: render })
              ] }, index);
            } else {
              return null;
            }
          } else {
            const args = message.arguments;
            let status = "inProgress";
            if (functionResults[message.id] !== void 0) {
              status = "complete";
            } else if (message.status.code !== MessageStatusCode.Pending) {
              status = "executing";
            }
            const toRender = render({
              status,
              args,
              result: functionResults[message.id]
            });
            if (!toRender && status === "complete") {
              return null;
            }
            if (typeof toRender === "string") {
              return /* @__PURE__ */ jsxs("div", { className: `copilotKitMessage copilotKitAssistantMessage`, children: [
                isCurrentMessage && inProgress && context.icons.spinnerIcon,
                " ",
                toRender
              ] }, index);
            } else {
              return /* @__PURE__ */ jsx("div", { className: "copilotKitCustomAssistantMessage", children: toRender }, index);
            }
          }
        } else if (!inProgress || !isCurrentMessage) {
          return null;
        } else {
          return /* @__PURE__ */ jsx("div", { className: `copilotKitMessage copilotKitAssistantMessage`, children: context.icons.spinnerIcon }, index);
        }
      } else if (message instanceof AgentStateMessage) {
        let render;
        if (chatComponentsCache.current !== null) {
          render = chatComponentsCache.current.coagentActions[`${message.agentName}-${message.nodeName}`] || chatComponentsCache.current.coagentActions[`${message.agentName}-global`];
        }
        if (render) {
          if (typeof render === "string") {
            if (isCurrentMessage && inProgress) {
              return /* @__PURE__ */ jsxs("div", { className: `copilotKitMessage copilotKitAssistantMessage`, children: [
                context.icons.spinnerIcon,
                " ",
                /* @__PURE__ */ jsx("span", { className: "inProgressLabel", children: render })
              ] }, index);
            } else {
              return null;
            }
          } else {
            const state = message.state;
            let status = message.active ? "inProgress" : "complete";
            const toRender = render({
              status,
              state,
              nodeName: message.nodeName
            });
            if (!toRender && status === "complete") {
              return null;
            }
            if (typeof toRender === "string") {
              return /* @__PURE__ */ jsxs("div", { className: `copilotKitMessage copilotKitAssistantMessage`, children: [
                isCurrentMessage && inProgress && context.icons.spinnerIcon,
                " ",
                toRender
              ] }, index);
            } else {
              return /* @__PURE__ */ jsx("div", { className: "copilotKitCustomAssistantMessage", children: toRender }, index);
            }
          }
        } else if (!inProgress || !isCurrentMessage) {
          return null;
        } else {
          return /* @__PURE__ */ jsx("div", { className: `copilotKitMessage copilotKitAssistantMessage`, children: context.icons.spinnerIcon }, index);
        }
      } else if (message instanceof ResultMessage && inProgress && isCurrentMessage) {
        return /* @__PURE__ */ jsx("div", { className: `copilotKitMessage copilotKitAssistantMessage`, children: context.icons.spinnerIcon }, index);
      }
    }),
    /* @__PURE__ */ jsx("footer", { ref: messagesEndRef, children })
  ] });
};
function makeInitialMessages(initial) {
  let initialArray = [];
  if (initial) {
    if (Array.isArray(initial)) {
      initialArray.push(...initial);
    } else {
      initialArray.push(initial);
    }
  }
  return initialArray.map(
    (message) => new TextMessage({
      role: Role.Assistant,
      content: message
    })
  );
}

export {
  Messages
};
