// @copilotkit/react-ui/dist/chunk-B2KQEX2R.mjs
import {
  Window
} from "./chunk-FO7Z5LAL.mjs";
import {
  Header
} from "./chunk-FLV3J3VX.mjs";
import {
  Button
} from "./chunk-YOH25I6N.mjs";
import {
  CopilotChat
} from "./chunk-HR36Y2FQ.mjs";
import {
  Input
} from "./chunk-U6J5DGOE.mjs";
import {
  Messages
} from "./chunk-FOZVHL73.mjs";
import {
  ResponseButton
} from "./chunk-XSUSSWDS.mjs";
import {
  ChatContextProvider
} from "./chunk-BJPGMY3I.mjs";

// src/components/chat/Modal.tsx
import React from "react";
import { jsx, jsxs } from "react/jsx-runtime";
var CopilotModal = ({
  instructions,
  defaultOpen = false,
  clickOutsideToClose = true,
  hitEscapeToClose = true,
  onSetOpen,
  onSubmitMessage,
  shortcut = "/",
  icons,
  labels,
  makeSystemMessage,
  showResponseButton = true,
  onInProgress,
  Window: Window2 = Window,
  Button: Button2 = Button,
  Header: Header2 = Header,
  Messages: Messages2 = Messages,
  Input: Input2 = Input,
  ResponseButton: ResponseButton2 = ResponseButton,
  className,
  children
}) => {
  const [openState, setOpenState] = React.useState(defaultOpen);
  const setOpen = (open) => {
    onSetOpen == null ? void 0 : onSetOpen(open);
    setOpenState(open);
  };
  return /* @__PURE__ */ jsxs(ChatContextProvider, { icons, labels, open: openState, setOpen, children: [
    children,
    /* @__PURE__ */ jsxs("div", { className, children: [
      /* @__PURE__ */ jsx(Button2, {}),
      /* @__PURE__ */ jsxs(
        Window2,
        {
          clickOutsideToClose,
          shortcut,
          hitEscapeToClose,
          children: [
            /* @__PURE__ */ jsx(Header2, {}),
            /* @__PURE__ */ jsx(
              CopilotChat,
              {
                instructions,
                onSubmitMessage,
                makeSystemMessage,
                showResponseButton,
                onInProgress,
                Messages: Messages2,
                Input: Input2,
                ResponseButton: ResponseButton2
              }
            )
          ]
        }
      )
    ] })
  ] });
};

export {
  CopilotModal
};
